chain/network/src/peer/peer_actor.rs

```diff
@@ -1303,7 +1316,7 @@ impl PeerActor {
                     return;
                 }
 
-                self.network_state.tier2_add_route_back(&self.clock, &conn, msg.as_ref());
+                self.network_state.add_route_back(&self.clock, &conn, msg.as_ref());
```

Comment:
Good question. The historical purpose of the RouteBackCache is to allow nodes to join and use the network smoothly (see https://github.com/near/nearcore/issues/2989). When a node joins it quickly learns from its neighbors how to route to targets in the network. It takes longer for the network as a whole to learn how to route back to the new node. The RouteBackCache helps to avoid dropping responses during that gap.

For the future, we also plan to add some optimization in which nodes only compute routes/distances to validators. The RouteBackCache will be even more important then for routing responses back to arbitrary nodes.